### PR TITLE
Hawkowl/proxy rhel8

### PR DESCRIPTION
**Pull request by hawkowl**
### Which issue this PR addresses:

Fixes https://issues.redhat.com/browse/ARO-2526

### What this PR does / why we need it:
Updates the proxy to run on RHEL8, and also has a multi-stage image because I'm scared that my Fedora box will have a different glibc :)

### Test plan for issue:
E2E infra, so, if the e2e still works?

### Is there any documentation that needs to be updated for this PR?
N/A

### CODE REVIEW

**Review by cblecker — APPROVED** [review]

File: Dockerfile.proxy [hunk 1]
```diff
@@ -1,7 +1,17 @@
+# Uses a multi-stage container build to build the proxy
+#
 ARG REGISTRY
+FROM ${REGISTRY}/ubi8/go-toolset:1.18.4 AS builder
+ENV GOOS=linux \
+    GOPATH=/go/
+WORKDIR ${GOPATH}/src/github.com/Azure/ARO-RP
+USER root
+COPY . ${GOPATH}/src/github.com/Azure/ARO-RP/
+RUN make proxy RELEASE=${IS_OFFICIAL_RELEASE} -o generate
```
Comment (cblecker) [inline Dockerfile.proxy:10]: Does this need to be an ARG too?

**Review by cblecker — CHANGES_REQUESTED** [review]

**Review by cblecker — DISMISSED** [review]

**Review by ellis-johnson — APPROVED** [review]
LGTM!

**Review by SrinivasAtmakuri — APPROVED** [review]
LGTM!

**Review by bennerv — APPROVED** [review]
lgtm

**Review by cadenmarchese — APPROVED** [review]

**Pull request merged**